### PR TITLE
Fix for Issue #4660. 

### LINKED ISSUE #4660: vector inverse for epochs data
A couple of issues with running the tutorial plot_mne_dspm_source_localization with epoch data instead of evoked. Specifically, I am trying to compute a vector inverse using the line:
`stc_vec = mne.minimum_norm.apply_inverse_epochs(epochs[0], inverse_operator, lambda2, method=method, nave=1, pick_ori='vector', return_generator=False)[0]`

Upon further inspection I found some apparent bugs:
1. In line [1067](https://github.com/mne-tools/mne-python/blob/master/mne/minimum_norm/inverse.py#L1067) of inverse.py it compares `len(sel)` to `K.shape[0]` when in fact it should compare to `K.shape[1]`, the channel axis
2. In line [1042](https://github.com/mne-tools/mne-python/blob/master/mne/minimum_norm/inverse.py#L1042) of inverse.py it should not require that `pick_ori is None`, since later in line 1060 it checks for `pick_ori != 'vector'` (and behaves correctly when `pick_ori==vector`). 
3. If `_make_stc` is called with a tuple (which happens whenever line [1068](https://github.com/mne-tools/mne-python/blob/master/mne/minimum_norm/inverse.py#L1068) is executed), it breaks because it tries to access `data.shape`

I can compute the inverse with corrections (1) and/or (2), but I think someone more familiar with the intended logic should review it to make sure it makes sense.

### PR DESCRIPTION
**Pull request by emilyps14**
Correcting bugs in _apply_inverse_epochs_gen, and adding a test to make sure apply_inverse and apply_inverse_epochs are consistent.

I see that the testing data doesn't have any *-epo.fif. The test here creates an Epochs on the fly from sample_audvis_trunc_raw.fif, and compares apply_inverse to apply_inverse_epochs with it. Would it be better to save the Epochs to the testing dataset and load it in the test?

Closes #4660

### CODE REVIEW

**Comment by larsoner** [comment]
> Would it be better to save the Epochs to the testing dataset and load it in the test?

Nah, the epoching operation should be pretty fast, and we want to keep the `testing` dataset size as small as possible.

File: mne/tests/test_source_estimate.py [hunk 1]
```diff
@@ -886,6 +889,39 @@ def test_vec_stc():
...
+    lambda2 = 1. / snr ** 2
+
+    stcs_epo = apply_inverse_epochs(epochs, inv, lambda2, method=method,
+                                    pick_ori='vector', return_generator=False)
+    stc_epo = np.mean(stcs_epo)
+
+    stc_evo = apply_inverse(evoked, inv, lambda2, method=method,
+                            pick_ori='vector')
+
+    assert(np.allclose(stc_epo.data,stc_evo.data))
```
Comment (larsoner) [inline mne/tests/test_source_estimate.py]: Just use `assert_allclose` here from `numpy.testing` (it is probably already imported)

Also check to see what values of `rtol` and `atol` can be used -- set as small as possible (to an order of magnitude) that will still pass, and ensure it makes sense relative to the scale of the actual data

File: mne/tests/test_source_estimate.py [hunk 2]
```diff
@@ -886,6 +889,39 @@ def test_vec_stc():
     assert_array_equal(normal.data[:, 0], [1, 2, 0, np.sqrt(3)])
 
 
+def test_epochs_vector_inverse():
+    """Test vector inverse consistency between evoked and epochs"""
+
+    raw = read_raw_fif(fname_raw)
+    events = find_events(raw, stim_channel='STI 014')
+    event_id = dict(aud_l=1)  # event trigger and conditions
```
Comment (larsoner) [inline mne/tests/test_source_estimate.py]: can just use `event_id=None` to take whatever events come first

File: mne/tests/test_source_estimate.py [hunk 3]
```diff
@@ -886,6 +889,39 @@ def test_vec_stc():
     assert_array_equal(normal.data[:, 0], [1, 2, 0, np.sqrt(3)])
 
 
+def test_epochs_vector_inverse():
+    """Test vector inverse consistency between evoked and epochs"""
+
+    raw = read_raw_fif(fname_raw)
+    events = find_events(raw, stim_channel='STI 014')
```
Comment (larsoner) [inline mne/tests/test_source_estimate.py]: use `events = ...[:2]` to just use two events for speed

File: mne/tests/test_source_estimate.py [hunk 4]
```diff
@@ -886,6 +889,39 @@ def test_vec_stc():
...
+    event_id = dict(aud_l=1)  # event trigger and conditions
+    tmin = -0.2  # start of each epoch (200ms before the trigger)
+    tmax = 0.5  # end of each epoch (500ms after the trigger)
+    raw.info['bads'] = ['MEG 2443', 'EEG 053']
+    baseline = (None, 0)  # means from the first instant to t = 0
+    reject = dict(grad=2000e-13, mag=4e-12, eog=150e-6)
+
+    epochs = Epochs(raw, events, event_id, tmin, tmax, baseline=baseline,
+                    reject=reject, preload=True).crop(0, 0.01)
```
Comment (larsoner) [inline mne/tests/test_source_estimate.py]: let's add:
```
assert len(epochs) == 2
```
to make sure we actually got more than one epoch

Comment (larsoner) [reply mne/tests/test_source_estimate.py]: ... and do `tmin=0., tmax=0.01, baseline=None` above instead of cropping here (the DC offset shouldn't matter anyway)

**Comment by emilyps14** [comment]
Also took out adding to raw.info['bads'], since it's correct on disk.

**Comment by larsoner** [comment]
Sounds good. FYI I added `Closes #4660` so that merging this PR will auto-close the related issue when it's merged.

File: mne/tests/test_source_estimate.py [hunk 5]
```diff
@@ -886,6 +889,36 @@ def test_vec_stc():
     assert_array_equal(normal.data[:, 0], [1, 2, 0, np.sqrt(3)])
 
 
+def test_epochs_vector_inverse():
```
Comment (larsoner) [inline mne/tests/test_source_estimate.py]: you need to add the `@testing.requires_testing_data` decorator here

**Comment by emilyps14** [comment]
I made a new commit with the decorator and changes to `whats_new`, but it isn't refreshing here?

**Comment by larsoner** [comment]
GitHub seems to be having a slow day, it has updated now.

File: doc/whats_new.rst [hunk 6]
```diff
@@ -208,8 +208,9 @@ BUG
 
     - Fix depth weighting of sparse solvers (:func:`mne.inverse_sparse.mixed_norm`, :func:`mne.inverse_sparse.tf_mixed_norm` and :func:`mne.inverse_sparse.gamma_map`) with free orientation source spaces to improve orientation estimation by `Alex Gramfort`_ and `Yousra Bekhti`_
 
-    - Fix the threshold in :func:`mne.beamformer.rap_music` to properly estimate the rank by `Yousra Bekhti`_
+    - Fix treatment of vector inverse in :func:`mne.minimum_norm.apply_inverse_epochs` by `Emily Stephen`_
 
+    - Fix the threshold in :func:`mne.beamformer.rap_music` to properly estimate the rank by `Yousra Bekhti`_
```
Comment (agramfort) [inline doc/whats_new.rst]: this line is unrelated

**Comment by agramfort** [comment]
besides LGTM

File: doc/whats_new.rst (hunk 6 above)
Comment (larsoner) [reply doc/whats_new.rst]: it looks like it's just a weird GitHub diff, I don't see any actual change

**Comment by larsoner** [comment]
Merged by squash+rebase after a flake fix, thanks @emilyps14
